logical/framework/field_type.go

```diff
@@ -84,3 +89,44 @@ func (t FieldType) String() string {
 		return "unknown type"
 	}
 }
+
+/*
+	The reason we wrap an http.Header here is - under the hood,
+	an http.Header is a map[string][]string and it does some magic
+	with casing that can make it difficult to correctly iterate
+	all header values.
+
+	For example:
+	h := http.Header{}
+	h.Add("hello", "world")
+	h.Add("hello", "monde")
+
+	You'd expect that the header now have "hello" for a key, and
+	both "world" and "monde" for a value. But it doesn't. "Hello"
+	is now capitalized in the map, but "world" isn't.
+
+	Later, when you do this:
+	h.Get("hello")
+
+	You'll receive only "world".
+
+	If you try to solve for this by doing this:
+	h["hello"]
+
+	You'll receive nothing back, because remember, it's now in the
+	map as "Hello.
+
+	To avoid bugs like this, we provide one more method. GetAll,
+	which returns all the values for a key.
+*/
+func NewHeader() Header {
```

Comment:
This should probably all go in parseutil if ParseHeader is in there (which it may be?)

Reply:
Np, I'll move it!